### PR TITLE
ci: update linter job and add bodyclose

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

- Remove deprecated lint job
- Make all teams own linter rules
- Add bodyclose linter rule
- Apply bodyclose fixes

### Motivation

- Remove deprecated tooling
- Safer code by adding the bodyclose rule
- All teams own the linter rules to make sure every team approves any future changes in the linter rules

### Describe how to test/QA your changes

The CI passes

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

File: contrib/twitchtv/twirp/twirp_test.go
Comment (ahmed-mez) [inline contrib/twitchtv/twirp/twirp_test.go:1]: In this contrib `wc.Do(req)` returns a nil error **and** a nil response, that's why it's skipped.

**Review by ajgajg1134 — APPROVED** [review]
Looks good! 🎉

File: contrib/net/http/roundtripper_test.go [hunk 1]
```diff
@@ -205,7 +210,8 @@ func TestRoundTripperNetworkErrorWithErrorCheck(t *testing.T) {
 			Timeout:   1 * time.Millisecond,
 		}
 
-		client.Get(s.URL + "/hello/world")
+		_, err := client.Get(s.URL + "/hello/world") //nolint:bodyclose
```
Comment (rarguelloF) [inline contrib/net/http/roundtripper_test.go:213]: (same question here)

File: internal/httpmem/httpmem_test.go [hunk 2]
```diff
@@ -31,7 +33,10 @@ func TestServerClosed(t *testing.T) {
...
 		t.Fatal(err)
 	}
-	if _, err := c.Do(r); err == nil {
-		t.Fatal("request should have failed, but it succeeded")
+	resp, err := c.Do(r)
+	if err != nil {
+		return
 	}
+	defer resp.Body.Close()
+	t.Fatal("request should have failed, but it succeeded")
```
Comment (rarguelloF) [inline internal/httpmem/httpmem_test.go:41]:
```suggestion
	_, err = c.Do(r)
	require.Error(t, err, "request should have failed, but it succeeded")
```
would the linter force you to close the body in this case?

**Review by rarguelloF — COMMENTED** [review]
Thanks for the PR!! 🎉 
LGTM, left some questions / small suggestion

File: contrib/net/http/roundtripper_test.go [hunk 3]
```diff
@@ -166,7 +170,8 @@ func TestRoundTripperNetworkError(t *testing.T) {
 		Timeout:   1 * time.Millisecond,
 	}
 
-	client.Get(s.URL + "/hello/world")
+	_, err := client.Get(s.URL + "/hello/world") //nolint:bodyclose
```
[the remark this answers is not in the document]
Comment (ahmed-mez) [reply contrib/net/http/roundtripper_test.go:173]: We expect a non-nil error, so the response is always nil.

File: contrib/net/http/roundtripper_test.go (hunk 1 above)
Comment (ahmed-mez) [reply contrib/net/http/roundtripper_test.go:213]: same answer

File: internal/httpmem/httpmem_test.go (hunk 2 above)
Comment (ahmed-mez) [reply internal/httpmem/httpmem_test.go:41]: Yes!

Comment (ahmed-mez) [reply internal/httpmem/httpmem_test.go:41]: See https://stackoverflow.com/questions/18598780/is-resp-body-close-necessary-if-we-dont-read-anything-from-the-body

**Review by Hellzy — APPROVED** [review]
Looking good on the AppSec side, thanks for this @ahmed-mez 👍

**Pull request merged**